src/viem-config.ts

```diff
@@ -0,0 +1,29 @@
+import { Chain, defineChain } from 'viem';
+import { CHAINS } from './generated/chains';
+
+export function chains(): Chain[] {
+  return CHAINS.map((chain) => {
+    const viemChain = defineChain({
+      id: Number(chain.id),
+      name: chain.name,
+      network: chain.alias,
+      nativeCurrency: chain.nativeCurrency,
+      rpcUrls: {
+        default: {
+          http: [chain.providerUrl],
+        },
+        public: {
+          http: [chain.providerUrl],
+        },
+      },
+      blockExplorers: {
+        default: {
+          name: 'Explorer',
+          url: chain.explorer.browserUrl,
+        },
+      },
```

Comment:
I'm not quite sure I understand the difference between a "default" provider and a "public" provider. Also related to this: https://github.com/api3dao/chains/pull/108#discussion_r1368379592

Reply:
To my understanding `public` is the URL of the official RPC that "the team" provides, and `default` is what we recommend the user to use. `default` must exist, while `public` may not exist (e.g. Ethereum). In a lot of the cases where `public` exists, it will also be `default`, except we sometimes recommend a third-party RPC (e.g. https://github.com/api3dao/chains/blob/ee80000efc629d4f646fe67c6bf00e2a83bf707b/chains/polygon.json#L7)

Reply:
So in this case, I would only do
```
      rpcUrls: {
        default: {
          http: [chain.providerUrl],
        },
      },
```

Reply:
Turns out viem requires both `default` and `public` as part of the `rpcUrls` object